test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -735,5 +735,115 @@ public void UninstallDeprecatedPackageFromUI(ProjectTemplate projectTemplate, st
+            var project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.NetCoreClassLib, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
+
+            var uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
+            solutionService.Build();
+
+            CommonUtility.AssertPackageReferenceExists(VisualStudio, project, TestPackageName, TestPackageVersionV1, Logger);
+            uiwindow.AssertTopLevelPackageExistsInInstalledTab(TestPackageName);
+            uiwindow.AssertTransitivePackageExistsInInstalledTab(transitivePackageName);
+
+            // Act
+            uiwindow.SearchPackageFromUI(transitivePackageName);
+
+            // Assert
+            VisualStudio.AssertNoErrors();
+            uiwindow.AssertSearchedTransitivePackageExistsInInstalledTab(transitivePackageName);
+        }
+
+        [TestMethod]
+        [Timeout(DefaultTimeout)]
+        public async Task InstallTransitivePackageFromUI()
+        {
+            // Arrange
+            var transitivePackageName = "Contoso.B";
+            await CommonUtility.CreateDependenciesPackageInSourceAsync(_pathContext.PackageSource, TestPackageName, TestPackageVersionV1, transitivePackageName, TestPackageVersionV1);
+
+            NuGetApexTestService nugetTestService = GetNuGetTestService();
+
+            var solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            var project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.NetCoreConsoleApp, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
+
+            var uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
+            solutionService.Build();
+
+            uiwindow.AssertTopLevelPackageExistsInInstalledTab(TestPackageName);
+            uiwindow.AssertTransitivePackageExistsInInstalledTab(transitivePackageName);
+
+            // Act
+            uiwindow.InstallPackageFromUI(transitivePackageName, TestPackageVersionV1);
+            solutionService.Build();
+
+            // Assert
+            VisualStudio.AssertNoErrors();
+            CommonUtility.AssertPackageReferenceExists(VisualStudio, project, transitivePackageName, TestPackageVersionV1, Logger);
+            uiwindow.AssertTransitivePackageNotExistsInInstalledTab(transitivePackageName);
+        }
+
+        [TestMethod]
+        [Timeout(DefaultTimeout)]
+        public async Task UninstallTopLevelToTransitivePackageFromUI()
```

Comment:
I mentioned earlier that we need a better name to explain the test case, since this does a lot. 

Uninstall_WithMultiplePackagesThatDependOnEachOther_PackageGoesFromDirectToTransitive


this is wordy, but it's important that test name describe the test case.